rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/analysis/AnalysisRpcService.java

```diff
@@ -104,4 +104,40 @@ public interface AnalysisRpcService {
    */
   @JsonNotification
   void didChangeAutomaticAnalysisSetting(DidChangeAutomaticAnalysisSettingParams params);
+
+  /**
+   * Analyze all files in the project. User file exclusions and .gitignore will be respected.
+   * @param params configuration scope ID, flag to report only hotspots
+   * Issues will be reported to the client via
+   * {@link SonarLintRpcClient#raiseIssues(RaiseIssuesParams)} and {@link SonarLintRpcClient#raiseHotspots(RaiseHotspotsParams)}
+   */
+  @JsonNotification
+  void analyzeFullProject(AnalyzeFullProjectParams params);
```

Comment:
In the ticket, we mentioned all these methods should be requests, and they should return at least an ID, so that clients can connect the dots when issues are raised later

Reply:
Changed as discussed.